packages/wonder-blocks-tooltip/components/tooltip-arrow.js

```diff
@@ -94,73 +119,136 @@ export default class TooltipArrow extends React.Component<Props> {
         }
     }
 
+    _getFilterPositioning() {
+        const {placement} = this.props;
+        switch (placement) {
+            case "top":
+                return {
+                    height: "150%",
+                    y: "1%",
+                    width: "200%",
+                    x: "-50%",
+                };
+
+            case "bottom":
+                // No shadow on the arrow as it falls "under" the bubble.
+                return null;
+
+            case "left":
+                return {
+                    height: "150%",
+                    y: "50%",
+                    width: "105%",
+                    x: "1%",
+                };
+
+            case "right":
+                return {
+                    height: "150%",
+                    y: "50%",
+                    width: "99%",
+                    x: undefined,
+                };
+
+            default:
+                throw new Error(`Unknown placement: ${placement}`);
+        }
+    }
+
+    /**
+     * Create an SVG filter that applies a blur to an element.
+     * We'll apply it to a dark shape outlining the tooltip, which
+     * will produce the overall effect of a drop-shadow.
+     *
+     * Also, scope its ID by side, so that tooltips with other
+     * "side" values don't end up using the wrong filter from
+     * elsewhere in the document. (The `height` value depends on
+     * which way the arrow is turned!)
```

Comment:
It's possible that I don't know what `width` and `height` do in the filter.  I assume that they controlled the spread of the gaussian blur.  Is that not the case?

Reply:
I'm so surprised that you have to set the `width` and `height` explicitly on a filter.  SVG already has other ways to clip things.  I wonder what purpose this is serving.  🤔  No action required.

Reply:
I guess when it comes to some filters like gaussians the result being rendered could be quite big so forcing people to pick a clipping rect probably makes sense from a perf point of view.  The dev ergonomics are not the greatest though.